parsers/deathloopvaccine.py

```diff
@@ -0,0 +1,286 @@
+
+        # cut off the leading date-time stamp info
+        trunc_line = line[27:]
+
+        # does this line contain a death message
+        slain_regexp = r'^You have been slain'
+        m = re.match(slain_regexp, trunc_line)
+        if m:
+            # add this message to the list of death messages
+            self._death_list.append(line)
+            starprint(f'DeathLoopVaccine:  Death count = {len(self._death_list)}')
+
+        # a way to test - send a tell to death_loop
+        slain_regexp = r'^death_loop'
+        m = re.match(slain_regexp, trunc_line)
+        if m:
+            # add this message to the list of death messages
+            # since this is just for testing, disarm the kill-gun
+            self._death_list.append(line)
+            starprint(f'DeathLoopVaccine:  Death count = {len(self._death_list)}')
+            self._kill_armed = False
+
+        # only do the list-purging if there are already some death messages in the list, else skip this
+        if len(self._death_list) > 0:
+
+            # create a datetime object for this line, using the very capable datetime.strptime()
+            now = datetime.strptime(line[0:26], '[%a %b %d %H:%M:%S %Y]')
+
+            # now purge any death messages that are too old
+            done = False
+            while not done:
+                # if the list is empty, we're done
+                if len(self._death_list) == 0:
+                    self.reset()
+                    done = True
+                # if the list is not empty, check if we need to purge some old entries
+                else:
+                    oldest_line = self._death_list[0]
+                    oldest_time = datetime.strptime(oldest_line[0:26], '[%a %b %d %H:%M:%S %Y]')
+                    elapsed_seconds = now - oldest_time
+
+                    if elapsed_seconds.total_seconds() > self.deathloop_seconds:
+                        # that death message is too old, purge it
+                        self._death_list.pop(0)
+                        starprint(f'DeathLoopVaccine:  Death count = {len(self._death_list)}')
+                    else:
+                        # the oldest death message is inside the window, so we're done purging
+                        done = True
+
+    def check_not_afk(self, timestamp: datetime, text: str) -> None:
+        """
+        check for "proof of life" indications the player is really not AFK
+
+        Args:
+            timestamp: A datetime.datetime object, created from the timestamp text of the raw logfile line
+            text: The text following the everquest timestamp
+
+        Returns:
+            None:
+        """
+
+        # reconstruct the full logfile line
+        # this is a bit counter-intuitive, but the rest of the logic in this function was
+        # developed assuming the line was the full line, including the time stamp
+        line = f'[{timestamp.strftime("%a %b %d %H:%M:%S %Y")}] ' + text
```

Comment:
Again, not totally following why this is necessary to do, if you have a timestamp you can just use it, there's only one line involved here and it looks like half the time you're truncating this stuff off anyway

Reply:
see above